src/rgw/rgw_lc.cc

```diff
@@ -2069,46 +2070,46 @@ int RGWLC::process(int index, int max_lock_secs, LCWorker* worker,
 
     /* if there is nothing at head, try to reinitialize head.marker with the
      * first entry in the queue */
-    if (head.marker.empty() &&
-	allow_shard_rollover(cct, now, head.shard_rollover_date) /* prevent multiple passes by diff.
+    if (head->marker().empty() &&
+	allow_shard_rollover(cct, now, head->shard_rollover_date()) /* prevent multiple passes by diff.
 								  * rgws,in same cycle */) {
 
       ldpp_dout(this, 5) << "RGWLC::process() process shard rollover lc_shard=" << lc_shard
-			 << " head.marker=" << head.marker
-			 << " head.shard_rollover_date=" << head.shard_rollover_date
+			 << " head.marker=" << head->marker()
+			 << " head.shard_rollover_date=" << head->shard_rollover_date()
 			 << dendl;
 
-      vector<rgw::sal::Lifecycle::LCEntry> entries;
-      int ret = sal_lc->list_entries(lc_shard, head.marker, 1, entries);
+      vector<std::unique_ptr<rgw::sal::Lifecycle::LCEntry>> entries;
+      int ret = sal_lc->list_entries(lc_shard, head->marker(), 1, entries);
       if (ret < 0) {
 	ldpp_dout(this, 0) << "RGWLC::process() sal_lc->list_entries(lc_shard, head.marker, 1, "
 			   << "entries) returned error ret==" << ret << dendl;
 	goto exit;
       }
       if (entries.size() > 0) {
-	entry = entries.front();
-	head.marker = entry.bucket;
-	head.start_date = now;
-	head.shard_rollover_date = 0;
+	entry = std::move(entries.front());
+	head->marker() = entry->bucket();
+	head->start_date() = now;
```

Comment:
Is (e.g.) head->marker of type std::string?  I'm not used to seeing something like "head->marker()" on the left hand side of an assignment.  Can you recap what is happening here and why?

Reply:
(standin for all the other places where this happens)

Reply:
oh, it's a non-const reference, per discussion;  I guess I slightly favor that?  I mean, it's not common lisp here...

Reply:
It's a accessor, rather than a getter/setter pair.  I did it because it was quick and easy, and forgot about it.  I'll go back and fix it to use getter/setter instead.